src/Symfony/Component/Asset/UrlPackage.php

```diff
@@ -119,17 +120,50 @@ protected function chooseBaseUrl($path)
         return (int) fmod(hexdec(substr(hash('sha256', $path), 0, 10)), count($this->baseUrls));
     }
 
-    private function getSslUrls($urls)
+    /**
+     * Set the baseUrls var depending on the context.
+     */
+    private function setCurrentBaseUrls()
     {
-        $sslUrls = array();
+        if (!empty($this->urlList['httpsUrl']) && $this->getContext()->isSecure()) {
+            $this->baseUrls = $this->urlList['httpsUrl'];
+        } elseif ($this->isStrictHttp) {
+            $this->baseUrls = $this->urlList['httpUrl'];
+        } else {
+            $this->baseUrls = $this->urlList['fullUrl'];
+        }
+    }
+
+    /**
+     * Split urls in two categories: http & https urls
+     * Some url can be found in both categories (// & https depending on $isStrictHttp option).
+     *
+     * @param array $urls
+     * @param bool  $isStrictHttp
+     *
+     * @return array
+     */
+    private function splitBaseUrl(array $urls, $isStrictHttp)
+    {
+        $urlList = array('httpUrl' => array(), 'httpsUrl' => array(), 'fullUrl' => array());
```

Comment:
Instead of this array of arrays, having 2 private properties `baseSecureUrls` and `baseUnsecureUrls`.